identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -201,6 +195,13 @@ public void onCharacteristicReadRequest(BluetoothDevice device, int requestId, i
         if (mCharacteristicIdentUuid != null
                 && characteristic.getUuid().equals(mCharacteristicIdentUuid)) {
             try {
+                // Here we know the connected device is a mDoc/mDoc reader
+                if (mCurrentConnection != null
+                        && !mCurrentConnection.getAddress().equals(device.getAddress())) {
+                    Log.w(TAG, "Got a characteristic read request from a different device");
+                    return;
+                }
+                mCurrentConnection = device;
```

Comment:
I think it's better to actually defer setting `mCurrentConnection` until the GATT client writes `0x01` to the state characteristic... after all, the standard says _Data retrieval shall start by signalling the ‘Start’ value to the ‘State’ characteristic._ ... thanks for the detective work, I'll whip up a pull request that does this.